udata/harvest/backends/base.py

```diff
@@ -127,92 +128,92 @@ def has_feature(self, key):
     def get_filters(self):
         return self.config.get('filters', [])
 
-    def harvest(self):
-        '''Start the harvesting process'''
-        if self.perform_initialization() is not None:
-            self.process_items()
-            self.finalize()
-        return self.job
 
-    def perform_initialization(self):
-        '''Initialize the harvesting for a given job'''
-        log.debug('Initializing backend')
+    """
+    Parent class that wrap children methods to add error management and debug logs.
+
+    The flow is the following:
+        Parent                    Child
+
+        harvest         ->    inner_harvest()
+                                 /
+        process_dataset (create HarvestItem)  <------
+                    \
+                      --------> inner_process_dataset() (call get_dataset() and update object)
+
+
+        process_dataset:
+            1. Create HarvestItem
+            2. Call inner_process_dataset(item)
+            3. Save HarvestItem (dryrun)
+            4. Save dataset (dryrun)
+
+    """
+
+    def inner_harvest(self):
+        raise NotImplementedError
+    
+    def inner_process_dataset(self, item: HarvestItem) -> Dataset:
+        raise NotImplementedError
+
+    def harvest(self):
+        log.debug(f'Starting harvesting {self.source.name} ({self.source.url})…')
         factory = HarvestJob if self.dryrun else HarvestJob.objects.create
-        self.job = factory(status='initializing',
+        self.job = factory(status='initialized',
                            started=datetime.utcnow(),
                            source=self.source)
 
         before_harvest_job.send(self)
 
         try:
-            self.initialize()
-            self.job.status = 'initialized'
-            if not self.dryrun:
-                self.job.save()
+            self.inner_harvest()
+
+            if self.source.autoarchive:
+                self.autoarchive()
+
+            self.job.status = 'done'
+
+            if any(i.status == 'failed' for i in self.job.items):
+                self.job.status += '-errors'
         except HarvestValidationError as e:
```

Comment:
I'm unsure about this being raised? When does it happen?

Reply:
It happens inside the CKAN backend and the MAAF backend (to validate against a schema)